version: '3.3'

services:
  portainer:
    container_name: portainer
    image: portainer/portainer-ee:latest
    restart: unless-stopped
    ports:
      - 9000:9000
      - 9443:9443
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock
      - portainer-data:/data
    network_mode: bridge
    # Labels for Traefik
    # labels:
    #   - "traefik.enable=true"
    #   - "traefik.http.routers.portainer-secure.entrypoints=https"
    #   - "traefik.http.routers.portainer-secure.rule=Host(`$SITE_URL`)"
    #   - "traefik.http.routers.portainer-secure.tls=true"
    #   - "traefik.http.routers.portainer-secure.service=portainer"
    #   - "traefik.http.services.portainer.loadbalancer.server.port=9443"
    #   - "traefik.docker.network=proxy-traefik"
    #   - "traefik.http.routers.portainer.rule=Host(`$SITE_URL`)"
    #   - "traefik.http.routers.portainer.entrypoints=http"
    #   - "traefik.http.middlewares.portainer-https-redirect.redirectscheme.scheme=https"
    #   - "traefik.http.routers.portainer.middlewares=portainer-https-redirect"

volumes:
  portainer-data:
    external: true
    name: portainer-data

networks:
  default:
    external:
      name: shared